### ISSUE TITLE
bug: bar rendered outside the sizedBox

### ISSUE THREAD

**Issue by Dr-Blank** [suragch/audio_video_progress_bar#72]
```dart
SizedBox(
  height: 8,
  child: ProgressBar(
    progress: const Duration(minutes: 5),
    total: const Duration(minutes: 10),
    thumbRadius: 0,
    timeLabelLocation: TimeLabelLocation.none,
    thumbCanPaintOutsideBar: false,
    barHeight: 8,
  ),
),
```
![Screenshot_2024-05-17-17-20-54-402_com example](https://github.com/suragch/audio_video_progress_bar/assets/64108942/086d99fd-c9a1-47bc-b9a7-e476a0c65abe)

I am not sure if this the intended behavior or not, or if I am messing up something but the bar is centered at the start of the box instead of the center, hence half bar is outside the widget while half widget is empty.

**Comment by suragch** [comment]
Sorry, I can't really see what you mean from the image. Which part is the progress bar and which is the rest of the UI? Can you show another image that only shows the bar and the problem?

**Comment by Dr-Blank** [comment]
Sorry, I should have made it clear from the start.

### What I was trying to do
Use the progress bar as only a indicator like the in built linear progress bar. Hence I disabled the text labels and also zeroed out the thumb.

### The bug
When the ProgressBar is inside a sized box instead of rendering it inside the sized box, it spills out. The height it assumes is correct but it is translated up so that it overflows from the top.

### When?
This bug occurs only when `TimeLabelLocation.none` is used and `thumbRadius` is `0`.

### Steps to reproduce:

The progress bar is put inside a sized box of height 16.
```dart
SizedBox(
  height: 16,
  child: const ProgressBar(
    progress: Duration(minutes: 5),
    total: Duration(minutes: 10),
    thumbRadius: 0,
    barHeight: 16,
    timeLabelLocation: TimeLabelLocation.none,
  ),
),
```
Now inspecting the widget using devtools widget inspector we see the blue outline is where the bar should be laid out but if you see the bar (yellow color) is actually peeking out of the box. To be exact half of the bar is outside the box it should be inside of. So the height is correct, 16 px but 8 px is outside the box on top.

![image](https://github.com/suragch/audio_video_progress_bar/assets/64108942/7041919d-aad2-4b21-a838-92f228648f46)

![image](https://github.com/suragch/audio_video_progress_bar/assets/64108942/f4a1ba02-8d19-4369-a8de-a837c3fb9e89)

Here is the same thing with linear progress indicator.
```dart
SizedBox(
  height: 16,
  child: LinearProgressIndicator(
    value: 0.5,
    borderRadius: BorderRadius.circular(16),
  ),
),
```

See how the widget is completely inside the sized box and is not coming out of it from the top.
![image](https://github.com/suragch/audio_video_progress_bar/assets/64108942/bd3881c2-b3f4-47cf-8c10-227ae371972c)

![image](https://github.com/suragch/audio_video_progress_bar/assets/64108942/e9268f8c-3242-49a2-b68e-949abaeb94a6)

### Extra Notes

It is worth mentioning that bar height has nothing to do with this.

```dart
SizedBox(
  height: 16,
  child: const ProgressBar(
    progress: Duration(minutes: 5),
    total: Duration(minutes: 10),
    thumbRadius: 0,
    timeLabelLocation: TimeLabelLocation.none,
  ),
),
```
As you can see the widget is forced to be of height 16 but still half of the bar is outside the box
![image](https://github.com/suragch/audio_video_progress_bar/assets/64108942/912b830c-4df6-4fa9-8337-bde0b81bd6da)

Now lets increase the thumb radius to 8 instead of 0
```dart
SizedBox(
  height: 16,
  child: const ProgressBar(
    progress: Duration(minutes: 5),
    total: Duration(minutes: 10),
    thumbRadius: 8.0,
    timeLabelLocation: TimeLabelLocation.none,
  ),
),
```
And we see the bar is now inside the box.
![image](https://github.com/suragch/audio_video_progress_bar/assets/64108942/e078ecb0-9290-41b7-88c3-066566b179d0)

Also the sized box has nothing to do with this bug, here is one without a sized box. I was using it just to make sure the constraints are pass correctly.
```dart
const ProgressBar(
  progress: Duration(minutes: 5),
  total: Duration(minutes: 10),
  thumbRadius: 0,
  barHeight: 16,
  timeLabelLocation: TimeLabelLocation.none,
),
```
![image](https://github.com/suragch/audio_video_progress_bar/assets/64108942/cf6c7311-e2c3-4cd6-bea5-d48708865387)

And now with labels but no thumb radius
```dart
const ProgressBar(
  progress: Duration(minutes: 5),
  total: Duration(minutes: 10),
  thumbRadius: 0,
  barHeight: 16,
  timeLabelLocation: TimeLabelLocation.sides,
),
```
we see that the bar is now again centered and not overflowing.
![image](https://github.com/suragch/audio_video_progress_bar/assets/64108942/afa89c0d-2017-402b-8796-36e975b82f54)

**Comment by suragch** [comment]
Great explanation. That's much more clear. It looks like maybe I was getting the height from the thumb instead of the bar. I should probably take the max of the two. Let me take a look.

**Comment by suragch** [comment]
So it turned out that I was calculating the height from the 2 * thumb height when there are no labels.

**Comment by suragch** [comment]
This fix is published in version 2.0.3. Please test it and reopen if it still isn't working.
